controllers/users/users.js

```diff
@@ -60,3 +60,48 @@ exports.registerUser = (req, res) => {
     }
   });
 };
+
+exports.loginUser = (req, res) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res
+      .status(400)
+      .json({ errors: errors.array({ onlyFirstError: true }) });
+  }
+  const { password, email } = req.body;
+
+  User.findOne({ email: email.toLowerCase() }).then(user => {
+    if (!user) {
+      return res
+        .status(400)
+        .json({ errors: [{ msg: messages.INVALID_CREDENTIALS }] });
+    } else {
+      bcrypt.compare(password, user.password).then(isMatch => {
+        if (isMatch) {
+          const payload = {
```

Comment:
This is exactly the same code as we have in registration endpoint maybe we can extract it to one place.